File: flower/src/main/java/com/kharitonov/flower/repositories/UserRepository.java
package com.kharitonov.flower.repositories;

import com.kharitonov.flower.entities.UserEntity;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface UserRepository extends JpaRepository<UserEntity, Long> {

    Optional<UserEntity> findByLogin(String login);
}

File: src/main/java/by/hghgrtut/shop/repositories/UserRepository.kt
package by.hghgrtut.shop.repositories

import by.hghgrtut.shop.entities.UserEntity
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional

interface UserRepository : JpaRepository<UserEntity?, Long?> {

    fun findByLogin(login: String?): UserEntity?
}